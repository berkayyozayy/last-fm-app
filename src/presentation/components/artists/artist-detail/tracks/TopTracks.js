import { useLocation } from "react-router-dom";
import styles from "./TopTracks.module.css";
import { apiDataRoutes } from "../../../../../api-routes";
import useHttp from "../../../../hooks/use-http";
import Error from "../../../common/error/Error";
import Loading from "../../../common/loading/Loading";
import ListView from "../../../ui/list-view/ListView";

function TopTracks() {
  const location = useLocation();
  const params = new URLSearchParams(location.search);
  const name = params.get("name");
  const tracksUrl = apiDataRoutes.tracks(name);

  const { data, error, loading } = useHttp(tracksUrl);

  if (error) {
    return <Error message={error.message} />;
  }

  if (loading) {
    return <Loading />;
  }

  const trackList = data?.toptracks?.track;

  return (
    <div className={styles["tracks-wrapper"]}>
      {trackList.map((track) => {
        return (
          <ListView
            key={track.mbid + track.name}
            name={track.artist.name}
            label={track.name}
            count={track.playcount}
            image={track.image[2]["#text"]}
            showListeners={true}
            listeners={track.listeners}
          />
        );
      })}
    </div>
  );
}

export default TopTracks;
