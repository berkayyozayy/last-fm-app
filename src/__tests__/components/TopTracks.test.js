import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import TopTracks from "../../presentation/components/artists/artist-detail/tracks/TopTracks";

describe("TopTracks component", () => {
  test("it should render", () => {
    const { container } = render(
      <MemoryRouter>
        <TopTracks />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
