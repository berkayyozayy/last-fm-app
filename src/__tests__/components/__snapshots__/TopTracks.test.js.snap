// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTracks component it should render 1`] = `
<div>
  <div
    class="loading"
  >
    <div
      class="circle"
    />
  </div>
</div>
`;
